tests/test_api_requests.py

```diff
@@ -325,3 +228,1149 @@ async def parse(self, response):
+            },
+            {},
+            {
+                "httpResponseBody": True,
+                "httpResponseHeaders": True,
+            },
+            [],
+        ),
+        (
+            {
+                "headers": {"User-Agent": ""},
+            },
+            {},
+            {
+                "httpResponseBody": True,
+                "httpResponseHeaders": True,
+            },
+            ["cannot be mapped"],
+        ),
+        (
+            {
+                "headers": {"User-Agent": DEFAULT_USER_AGENT},
+                "meta": {"zyte_api": {"browserHtml": True}},
+            },
+            {},
+            {
+                "browserHtml": True,
+                "httpResponseHeaders": True,
+            },
+            [],
+        ),
+        (
+            {
+                "headers": {"User-Agent": ""},
+                "meta": {"zyte_api": {"browserHtml": True}},
+            },
+            {},
+            {
+                "browserHtml": True,
+                "httpResponseHeaders": True,
+            },
+            ["cannot be mapped"],
+        ),
+        # You may update the ZYTE_API_UNSUPPORTED_HEADERS setting to remove
+        # headers that the customHttpRequestHeaders parameter starts supporting
+        # in the future.
+        (
+            {
+                "headers": {
+                    "Cookie": "",
+                    "User-Agent": "",
+                },
+            },
+            {
+                "ZYTE_API_UNSUPPORTED_HEADERS": ["Cookie"],
+            },
+            {
+                "httpResponseBody": True,
+                "httpResponseHeaders": True,
+                "customHttpRequestHeaders": [
+                    {"name": "User-Agent", "value": ""},
+                ],
+            },
+            [
+                "defines header b'Cookie', which cannot be mapped",
+            ],
+        ),
+        # You may update the ZYTE_API_BROWSER_HEADERS setting to extend support
+        # for new fields that the requestHeaders parameter may support in the
+        # future.
+        (
+            {
+                "headers": {"User-Agent": ""},
+                "meta": {"zyte_api": {"browserHtml": True}},
+            },
+            {
+                "ZYTE_API_BROWSER_HEADERS": {
+                    "Referer": "referer",
+                    "User-Agent": "userAgent",
+                },
+            },
+            {
+                "browserHtml": True,
+                "httpResponseHeaders": True,
+                "requestHeaders": {"userAgent": ""},
+            },
+            [],
+        ),
+        # BODY
+        # The body is copied into httpRequestBody, base64-encoded.
+        (
+            {
+                "body": "a",
+            },
+            {},
+            {
+                "httpResponseBody": True,
+                "httpResponseHeaders": True,
+                "httpRequestBody": "YQ==",
```

Comment:
Should we also ensure that the `httpRequestMethod` is not `GET` when the body is set?

Reply:
I believe it is technically possible for it to be. If it was not technically possible, it should be Scrapy itself complaining, not this plugin.

I am not sure whether or not Zyte API itself allows it, but if it does not, I think it may be best to let Zyte API be the one who complains; similarly to how I think we should allow `httpResponseBody` and `browserHtml` be combined, even though Zyte API does not currently support that.